### PR TITLE
Rework display none while disabled

### PR DESCRIPTION
**Pull request by TheColaber**
Extends #4924

Changes Tab.displayNoneWhileDisabled to create a style element on the page that has a display: none !important when the addon is disabled.

Hopefully this can get merged before any new addons uses the api...

### CODE REVIEW

File: addons/one-click-studio-invites/userscript.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default async function (
       button.disabled = true;
     }
 
-    addon.tab.displayNoneWhileDisabled(button, { display: "block" });
+    addon.tab.displayNoneWhileDisabled(button);
```
Comment (mxmou) [inline addons/one-click-studio-invites/userscript.js:44]: Same as above, but with `display: block`.

File: addons/vol-slider/userscript.js [hunk 2]
```diff
@@ -55,9 +55,7 @@ export default async function ({ addon, console }) {
   innerContainer.appendChild(icon);
   innerContainer.appendChild(slider);
   container.appendChild(innerContainer);
-  addon.tab.displayNoneWhileDisabled(container, {
-    display: "flex",
-  });
+  addon.tab.displayNoneWhileDisabled(container);
```
Comment (mxmou) [inline addons/vol-slider/userscript.js:58]: Same

**Review by mxmou — CHANGES_REQUESTED** [review]

File: addons/emoji-picker/userscript.js [hunk 3]
```diff
@@ -179,7 +179,7 @@ export default async function ({ addon, console, msg }) {
     emojiButtonText.textContent = "🙂︎";
     emojiButtonText.classList.add("sa-emoji-button");
     emojiButton.appendChild(emojiButtonText);
-    addon.tab.displayNoneWhileDisabled(emojiButton, { display: "inline-block" });
+    addon.tab.displayNoneWhileDisabled(emojiButton);
```
[the remark this answers is not in the document]
Comment (TheColaber) [reply addons/emoji-picker/userscript.js:182]: i tested this addon, and all the ones you mentioned below, and they seemed to work without adding any extra styles

[the remark this answers is not in the document]
Comment (mxmou) [reply addons/emoji-picker/userscript.js:182]: They work but the layout is slightly broken - for example, the volume slider isn't centered. Compare them with the stable version.

File: addons/emoji-picker/userstyle.css [hunk 4]
```diff
@@ -1,4 +1,5 @@
 .sa-emoji-picker {
+  display: inline-block;
```
Comment (mxmou) [inline addons/emoji-picker/userstyle.css:2]: `.sa-emoji-button-container` also needs this style. It currently looks like this in this PR:
image

On master:
image

**Review by mxmou — APPROVED** [review]

**Review by Samq64 — APPROVED** [review]

**Comment by TheColaber** [comment]
Updated new addon. Good to go.

**Pull request merged**